src/lib/transaction_snark/test/test_zkapp_update.ml

```diff
@@ -50,32 +52,34 @@ module Make (Input : Input_intf) = struct
           ~snapp_pk:(Public_key.compress new_kp.public_key) )
 
   let%test_unit "update a snapp account with proof" =
-    Quickcheck.test ~trials:1 U.gen_snapp_ledger
-      ~f:(fun ({ init_ledger; specs = _ }, new_kp) ->
-        let fee = Fee.of_nanomina_int_exn 1_000_000 in
-        let amount = Amount.of_mina_int_exn 10 in
-        let test_spec : Spec.t =
-          { sender = (new_kp, Mina_base.Account.Nonce.zero)
-          ; fee
-          ; fee_payer = None
-          ; receivers = []
-          ; amount
-          ; zkapp_account_keypairs = [ new_kp ]
-          ; memo
-          ; new_zkapp_account = false
-          ; snapp_update
-          ; current_auth = Permissions.Auth_required.Proof
-          ; call_data = Snark_params.Tick.Field.zero
-          ; events = []
-          ; actions = []
-          ; preconditions = None
-          }
-        in
-        U.test_snapp_update
-          ~snapp_permissions:
-            (U.permissions_from_update snapp_update ~auth:Proof)
-          test_spec ~init_ledger ~vk ~zkapp_prover
-          ~snapp_pk:(Public_key.compress new_kp.public_key) )
+    if is_update_verification_key then ()
```

Comment:
Shouldn't this be applicable for vk update as well? If set-verification-key permission is `Proof` and there is Proof auth, it should pass, right? I think we should continue testing this case because there is no type-level restriction to avoid `Proof` or `impossible` values for set-verification-key permission. Curious if we can add such a restriction?

Reply:
This is not applicable for vk update because we can not set the permission to be `Proof` auth.

Reply:
As for type-level restriction, I can try that using GADT.